### PR TITLE
[PhotonPoseEstimator] Check for CalibData before adding Tags when using SolvePNP method

### LINKED ISSUE #825: Check CalibData before creating target and tag arrays
https://github.com/PhotonVision/photonvision/blob/dcd917870cece9e34eb32bf7ddc068d52dc5aefa/photon-lib/src/main/java/org/photonvision/PhotonPoseEstimator.java#L360

Instead of checking if calibration data is present after collecting all the tags and their corners, it should be checked before as a simple optimization.

### PR DESCRIPTION
**Pull request by srimanachanta**
closes #825

### CODE REVIEW

**Comment by mcm001** [comment]
In what way does this optimize the function? Both a list of valid targets and valid calibration data are necessary preconditions for solvePNP being run

**Comment by srimanachanta** [comment]
> In what way does this optimize the function? Both a list of valid targets and valid calibration data are necessary preconditions for solvePNP being run

Yeah, more of just a reorder, because it checks if PNP is possible due to the calibration matrices before adding the poses to arrays, and checking the field layout for the tags, it could simplify some of the logic

**Review by mcm001 — APPROVED** [review]

File: photon-lib/src/main/java/org/photonvision/PhotonPoseEstimator.java [hunk 1]
```diff
@@ -365,48 +364,44 @@ private Optional<EstimatedRobotPose> multiTagPNPStrategy(
...
-            // no calibration data -- run fallback strategy instead
-            return update(result, cameraMatrixOpt, distCoeffsOpt, this.multiTagFallbackStrategy);
-        }
-        Matrix<N3, N3> cameraMatrix = cameraMatrixOpt.get();
-        Matrix<N5, N1> distCoeffs = distCoeffsOpt.get();
 
-        ArrayList<TargetCorner> visCorners = new ArrayList<>();
-        ArrayList<AprilTag> knownVisTags = new ArrayList<>();
+        if(!hasCalibData) {
+            return Optional.empty();
```
Comment (mcm001) [inline photon-lib/src/main/java/org/photonvision/PhotonPoseEstimator.java:378]: Needs to be `return update(result, cameraMatrixOpt, distCoeffsOpt, this.multiTagFallbackStrategy);`

File: photon-lib/src/main/native/cpp/photonlib/PhotonPoseEstimator.cpp [hunk 2]
```diff
@@ -359,11 +359,16 @@ std::optional<EstimatedRobotPose> PhotonPoseEstimator::MultiTagPnpStrategy(
...
   using namespace frc;
 
+  // Need at least 2 targets
   if (!result.HasTargets() || result.GetTargets().size() < 2) {
     return Update(result, std::nullopt, std::nullopt,
                   this->multiTagFallbackStrategy);
   }
 
+  if (!camMat || !distCoeffs) {
+    return std::nullopt;
```
Comment (mcm001) [inline photon-lib/src/main/native/cpp/photonlib/PhotonPoseEstimator.cpp:369]: Also needs to be changed to `return Update(result, std::nullopt, std::nullopt,
                  this->multiTagFallbackStrategy);`

**Review by mcm001 — CHANGES_REQUESTED** [review]

**Review by mcm001 — APPROVED** [review]

**Pull request merged**